src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -752,35 +779,66 @@ protected bool TakeStartLine(SocketInput input)
                 if (chFound == '?')
                 {
                     begin = scan;
-                    if (scan.Seek(ref _vectorSpaces) != ' ')
+                    if (scan.Seek(ref _vectorSpaces) == -1)
                     {
-                        return false;
+                        return RequestLineStatus.TargetIncomplete;
                     }
                     queryString = begin.GetAsciiString(scan);
                 }
 
+                if (pathBegin.GetLength(scan) == 0)
```

Comment:
I figured I can just do `if (pathBegin.Peek() == ' ')` - if there's a space there we don't even care what comes next, since there should only be one space between the method and the target.